File: week7-tdd-rails/morn_exe/anagramdetector/anagramdetector.rb
require 'pry'
#{`"listen"` and `["enlists", "google", "inlets", "banana"]`}

array = %w{enlists google inlets banana}

test_word = ("listen".scan /\w/).sort

def check_anagram string1, string2
  string1.sort == string2.sort
end

array.each do |word|
  letters = word.scan /\w/
  # binding.pry
  if check_anagram letters, test_word
    puts "#{letters.join}"
  end
end

File: week7-tdd-rails/morn_exe/anagramdetector/anagram_detector.js
array = ["enlists", "google", "inlets", "banana"]

test_word = "listen".split("").sort();

array.forEach(function(word) {
  letters = word.split("");

  /* solution 2 with function*/
  // if(findAnagram(letters, test_word)){
  //   console.log(word);
  // }

  /* Solution 2 */
  if(letters.sort().join("") === test_word.join("")){
    console.log(word);
  }
});

function findAnagram(string1, string2) {
  string1 = string1.sort();

  if(string1.length !== string2.length) {
    return false;
  } else {
    for(var i= 0; i < string1.length; i++) {
      if(string1[i] !== string2[i]) {
        return false;
      }
    }
    return true;
  }
}